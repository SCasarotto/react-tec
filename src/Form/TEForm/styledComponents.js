import styled from 'styled-components'

export const Form = styled.form`
	position: relative;
	width: 100%;
	margin: 0px auto;
	box-sizing: border-box;
`
