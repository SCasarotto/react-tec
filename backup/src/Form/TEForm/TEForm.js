import React from 'react'

import { Form } from './styledComponents'

const TEForm = (props) => {
	const { className = '', ...rest } = props
	return <Form className={`TEForm ${className}`} {...rest} />
}

export default TEForm
