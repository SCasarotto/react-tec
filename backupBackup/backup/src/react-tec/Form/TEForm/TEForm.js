//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import { Form } from './styledComponents'

const TEForm = (props) => {
	const { children, ...rest } = props

	return <Form {...rest}>{children}</Form>
}

TEForm.propTypes = {
	onSubmit: PropTypes.func,
}

export default TEForm
