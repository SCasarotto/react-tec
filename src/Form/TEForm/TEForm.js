//
//TE Version 0.3.0
//

import React from 'react'

import { Form } from './styledComponents'

const TEForm = (props) => <Form {...props} />

export default TEForm
