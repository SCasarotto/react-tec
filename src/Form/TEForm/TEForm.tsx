import React from 'react'

import { Form } from './styledComponents'

export interface TEFormProps
	extends React.DetailedHTMLProps<React.FormHTMLAttributes<HTMLFormElement>, HTMLFormElement> {
	ref?:
		| ((instance: HTMLFormElement | null) => void)
		| React.RefObject<HTMLFormElement>
		| null
		| undefined
}
export const TEForm: React.FC<TEFormProps> = React.forwardRef((props, ref) => {
	const { className = '', ...rest } = props
	return <Form className={`TEForm ${className}`} ref={ref} {...rest} />
})
