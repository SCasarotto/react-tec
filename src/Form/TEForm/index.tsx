import React from 'react'

import { Form } from './styledComponents'

export interface TEFormProps
	extends React.DetailedHTMLProps<React.FormHTMLAttributes<HTMLFormElement>, HTMLFormElement> {
	//TODO: Fix this. styled-components is weird with refs
	ref?: any
}
export const TEForm: React.FC<TEFormProps> = (props) => {
	const { className = '', ...rest } = props
	return <Form className={`TEForm ${className}`} {...rest} />
}
