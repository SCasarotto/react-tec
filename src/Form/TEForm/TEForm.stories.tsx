import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEForm, TEFormProps } from './TEForm'
import { TEButton } from '../TEButton'

export default {
	title: 'Form/TEForm',
	component: TEForm,
	argTypes: {},
} as Meta

const Template: Story<TEFormProps> = (args) => <TEForm {...args} />

export const Default = Template.bind({})
Default.args = {
	children: <TEButton type='submit'>Submit</TEButton>,
	onSubmit: (e) => {
		e.preventDefault()
		alert('Submitted!')
	},
}
