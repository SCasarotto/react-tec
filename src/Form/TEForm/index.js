import TEForm from './TEForm'

export default TEForm
