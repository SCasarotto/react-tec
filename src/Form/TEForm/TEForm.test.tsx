import React from 'react'
import ReactDOM from 'react-dom'
import { TEForm } from './TEForm'

describe('TEForm', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(<TEForm />, div)
		ReactDOM.unmountComponentAtNode(div)
	})
})
