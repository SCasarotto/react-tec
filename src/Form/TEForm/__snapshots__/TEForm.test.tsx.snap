// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEForm matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="sc-bdnylx gCsuFE TEForm "
  />
</DocumentFragment>
`;
